cranelift-codegen/src/verifier/mod.rs

```diff
@@ -1873,4 +1892,49 @@ mod tests {
 
         assert_err_with_msg!(errors, "instruction format");
     }
+
+    #[test]
+    fn bad_lane_index() {
```

Comment:
(light suggestion) I think a clif test would be nice and simple to detect this. For what it's worth, you can ask for expected failures in verifier tests.